homeassistant/components/devolo_home_control/config_flow.py

```diff
@@ -0,0 +1,89 @@
+"""Config flow to configure the devolo home control integration."""
+import logging
+
+from devolo_home_control_api.mydevolo import Mydevolo, WrongCredentialsError
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _login_data_valid(user, password):
+    """Validate the given login data."""
+    mydevolo = Mydevolo.get_instance()
+    mydevolo.user = user
+    mydevolo.password = password
+    mydevolo.uuid  # pylint: disable=pointless-statement
```

Comment:
What happens here?

Reply:
I looked into the library and this does a request.

Firstly, I would recommend not doing I/O in properties at all. That pattern is not  friendly to readers of the code. Actions should be explicit so use functions for I/O.

Secondly, since this does I/O we need to run this whole function, `_login_data_valid`, in the executor thread pool using `hass.async_add_executor_job`. The event loop must never block.

Reply:
Changed the API to get a clearer view, what happens here